api/document/serializers/doc_cursor.py

```diff
@@ -1,13 +1,34 @@
+from typing import Any, Dict, List
+
 from rest_framework import serializers
 
 from document.models import DocNode
 from document.serializers.content import (NestedAnnotationSerializer,
                                           nest_annotations)
 from document.serializers.meta import Meta, MetaSerializer
+from document.tree import DocCursor
+
+JsonDict = Dict[str, Any]
+
+
+class DocCursorField(serializers.Field):
+    def get_attribute(self, instance: DocCursor) -> DocCursor:
+        return instance
+
+
+class ChildrenField(DocCursorField):
+    def to_representation(self, instance: DocCursor) -> List[JsonDict]:
+        return DocCursorSerializer(
+            instance.children(), many=True,
+            context={**self.context, 'is_root': False},
+        ).data
```

Comment:
I've verified the type annotations here by adding `assert` statements into the code and then re-running the tests (since I'm unfamiliar with the code, I wanted to verify that my assumptions were correct).

Reply:
Yeah, it's a bit unexpected, but the `.data` field will return either serialized version of an object (an ordered dict of string fields to other ordered dicts or value types) _or_ a list of serialized objects if `many=True`.